openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/edgefilters/ch/DownwardSearchEdgeFilter.java

```diff
@@ -20,19 +20,50 @@
 
 public class DownwardSearchEdgeFilter extends CHLevelEdgeFilter {
 	protected final BooleanEncodedValue accessEnc;
+	private boolean useCore = false;
+	private boolean useCoreTurnRestrictions = false;
+	private boolean swap = false;
+	private int coreNodeLevel = -1;
 
 
 	public DownwardSearchEdgeFilter(CHGraph g, FlagEncoder encoder) {
 		super(g, encoder);
 		accessEnc = encoder.getAccessEnc();
 	}
 
+	public DownwardSearchEdgeFilter(CHGraph g, FlagEncoder encoder, boolean useCore) {
+		this(g, encoder);
+		this.useCore = useCore;
+		this.coreNodeLevel = maxNodes + 1;
+
+	}
+
+	public DownwardSearchEdgeFilter(CHGraph g, FlagEncoder encoder, boolean useCore, boolean useCoreTurnRestrictions, boolean swap) {
+		this(g, encoder, useCore);
+		if(useCore == false && useCoreTurnRestrictions == true)
+			throw new IllegalArgumentException("If turn restrictions in core should be respected, core must be respected, too.");
+		this.swap = swap;
+	}
+
 	@Override
 	public boolean accept(EdgeIteratorState edgeIterState) {
 		int adj = edgeIterState.getAdjNode(); 
-
+		if(useCore && !useCoreTurnRestrictions) {
+			if (baseNode >= maxNodes || adj >= maxNodes || baseNodeLevel < graph.getLevel(adj))
+				return swap ? edgeIterState.get(accessEnc) : edgeIterState.getReverse(accessEnc);
+			else
+				return false;
+		}
+		if(useCore && useCoreTurnRestrictions) {
+			if(baseNodeLevel == coreNodeLevel && graph.getLevel(adj) == coreNodeLevel + 1)
+				return false;
```

Comment:
Isn't this combined condition actually about making sure that a more general condition `baseNodeLevel < coreNodeLevel` is fulfilled in order to prevent matching to the subsequent condition `baseNodeLevel < graph.getLevel(adj)` in the special case of `baseNodeLevel == coreNodeLevel && graph.getLevel(adj) == coreNodeLevel + 1`? It's not clear at first glance and it actually took me a while to understand the objective behind it 🙈

Reply:
Please re-view with the changes I added in the last commit.

Reply:
Fixed, thanks!